### PR TITLE
Support for Facebooks HHVM

### PR DESCRIPTION
**Pull request by MDXDave**
Customers can use HHVM insted of PHP if admin enables it.
Its (Sub)domain based

### CODE REVIEW

File: install/updates/froxlor/0.9/update_0.9.inc.php [hunk 1]
```diff
@@ -2999,7 +2999,19 @@
 
 if (isFroxlorVersion('0.9.34-dev4')) {
 
-    showUpdateStep("Updating from 0.9.34-dev4 to 0.9.34 final");
+    showUpdateStep("Updating from 0.9.34-dev4 to 0.9.34-dev5", false);
```
Comment (d00p) [inline install/updates/froxlor/0.9/update_0.9.inc.php]: wrong place for DB update, people running >0.9.34-dev5 will never get these, please always add a new -dev version to the bottom of the update_0.9.inc.php file

File: lib/configfiles/jessie.xml [hunk 2]
```diff
@@ -4017,6 +4017,14 @@ aliases:     files
 						dirty work -->
 					<command><![CDATA[php {{const.FROXLOR_INSTALL_DIR}}/scripts/froxlor_master_cronjob.php --force]]></command>
 				</daemon>
+				<!-- HHVM -->
+				<daemon name="hhvm" title="HHVM (HipHop Virtual Machine)">
+				<install><![CDATA[sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 0x5a16e7281be7a449]]></install>
+          			<install><![CDATA[echo deb http://dl.hhvm.com/debian jessie main | sudo tee /etc/apt/sources.list.d/hhvm.list]]></install>
+          			<install><![CDATA[sudo apt-get update]]></install>
+          			<install><![CDATA[sudo apt-get install hhvm]]></install>
+				<command><![CDATA[ /usr/bin/hhvm --config /etc/hhvm/php.ini --config /etc/hhvm/server.ini --user www-data --mode daemon -vPidFile=/var/run/hhvm/pid]]></command>
```
Comment (d00p) [inline lib/configfiles/jessie.xml]: there should be a startstop-script / init.d script...

File: lng/german.lng.php [hunk 3]
```diff
@@ -1579,3 +1579,8 @@
 $lng['serversettings']['default_vhostconf']['description'] = 'Der Inhalt dieses Feldes wird direkt in den IP/Port-vHost-Container übernommen. '.$lng['admin']['specialsettings_replacements'].'<br /><strong>ACHTUNG:</strong> Der Code wird nicht auf Fehler geprüft. Etwaige Fehler werden also auch übernommen. Der Webserver könnte nicht mehr starten!';
 $lng['serversettings']['default_vhostconf_domain']['description'] = 'Der Inhalt dieses Feldes wird direkt in jeden Domain-vHost-Container übernommen. '. $lng['admin']['specialsettings_replacements'].'<strong>ACHTUNG:</strong> Der Code wird nicht auf Fehler geprüft. Etwaige Fehler werden also auch übernommen. Der Webserver könnte nicht mehr starten!';
 $lng['admin']['mod_fcgid_umask']['title'] = 'Umask (Standard: 022)';
+
+// HHVM Patch
+$lng['serversettings']['mod_fcgid']['hhvm'] = 'Kunden können HHVM statt PHP verwenden';
+$lng['domains']['hhvm']['title'] = "HHVM verwenden";
+$lng['domains']['hhvm']['description'] = "Auswählen um HHVM statt PHP zu verwenden.";
```
Comment (d00p) [inline lng/german.lng.php]: where are the language strings in english.lng.php - which is the default and fallback and should always be present

File: scripts/jobs/cron_tasks.inc.http.15.apache_fcgid.php [hunk 4]
```diff
@@ -93,16 +93,22 @@ protected function composePhpOptions($domain, $ssl_vhost = false)
 				else
 				{
 					$php_options_text.= '  SuexecUserGroup "' . $domain['loginname'] . '" "' . $domain['loginname'] . '"' . "\n";
+					if( $domain["hhvm"] && Settings::Get('system.mod_fcgid_hhvm') == 1 ){
+						$php_options_text.= '	ProxyPassMatch ^/(.*\.php(/.*)?)$ fcgi://127.0.0.1:9000/'.makeCorrectDir($domain['documentroot'])."\$1 \n";
```
Comment (d00p) [inline scripts/jobs/cron_tasks.inc.http.15.apache_fcgid.php]: you should make (at least) the port a configuration

**Comment by d00p** [comment]
Missing config-templates for other supported distributions; missing support for lighttpd and nginx

**Comment by unicco33** [comment]
I can't find this HHVM support in the demo. Where is this option be listed?

**Comment by nachtgeist** [comment]
This isn't merged so you won't find it.

**Comment by unicco33** [comment]
Thank you for your reply nachtgeist. Do you know the status for this merge? I'm in the offing to select a new Control Panel for a new server-configuration, and would like a CP that have HHVM implemented.

**Comment by d00p** [comment]
See https://github.com/Froxlor/Froxlor/pull/268#issuecomment-163519886

**Pull request closed without merging**